suite-native/transactions/src/components/TransactionName.tsx

```diff
@@ -0,0 +1,64 @@
+import { WalletAccountTransaction } from '@suite-common/wallet-types';
+import { Translation, TxKeyPath } from '@suite-native/intl';
+
+type TransactionNameProps = {
+    transaction: WalletAccountTransaction;
+    isPending: boolean;
+};
+
+interface GetSelfTransactionMessageByTypeProps {
+    type?: Required<WalletAccountTransaction>['cardanoSpecific']['subtype'];
+}
+
+const getSelfTransactionMessageByType = ({
+    type,
+}: GetSelfTransactionMessageByTypeProps): TxKeyPath => {
+    switch (type) {
+        case 'withdrawal':
+            return 'transactions.name.withdrawal';
+        case 'stake_delegation':
+            return 'transactions.name.stake_delegation';
+        case 'stake_registration':
+            return 'transactions.name.stake_registration';
+        case 'stake_deregistration':
+            return 'transactions.name.stake_deregistration';
+        default:
+            return 'transactions.name.self';
+    }
+};
+
+export const getTransactionName = (
+    transaction: WalletAccountTransaction,
+    isPending: boolean,
+): TxKeyPath => {
+    switch (transaction.type) {
+        case 'sent':
+            return isPending ? 'transactions.name.sending' : 'transactions.name.sent';
+        case 'recv':
+            return isPending ? 'transactions.name.receiving' : 'transactions.name.received';
+        case 'failed':
+            return 'transactions.name.failed';
+        case 'joint':
+            return 'transactions.name.joint';
+        case 'contract':
+            return 'transactions.name.contract';
+        case 'self':
+            return getSelfTransactionMessageByType({
+                type: transaction.cardanoSpecific?.subtype,
+            });
+        case 'unknown':
+        default:
+            return isPending ? 'transactions.name.pending' : 'transactions.name.unknown';
+    }
+};
+
+export const TransactionName = ({ transaction, isPending }: TransactionNameProps) => {
+    const ethName = transaction.ethereumSpecific?.parsedData?.name;
+
+    // use name of eth txns, but not for Transfer
+    if (ethName && !(ethName === 'Transfer' && ['recv', 'sent'].includes(transaction.type))) {
```

Comment:
Are you sure about the `&&` operator after `'Transfer'`? 🤔 

Anyway, I'd propagate the negation deeper. Since the original condition is off from my PoV, notice I kept the `&&` operators.
```suggestion
    if (ethName && (ethName !== 'Transfer' && transaction.type !== 'recv' && transaction.type !== 'sent')) {
```

Reply:
32c38f50ee

Reply:
ah dc0ee8c39f

Reply:
It should apply ethName only for txnst that are not Transfer AND (sent || recv)